### ISSUE TITLE
add more options for dev. board connections

### ISSUE THREAD

**Issue by rursprung** [tweedegolf/drive-rs#86]
currently only `StemmaQT` and `MikroBus` are allowed:
https://github.com/tweedegolf/drive-rs/blob/20b5d4517a4788f66c87e8df2a77a336b4a457dd/driver-db-schema.json#L153-L159

thus e.g. the ssd1306 dev. board is listed only as `StemmaQT`:
https://github.com/tweedegolf/drive-rs/blob/20b5d4517a4788f66c87e8df2a77a336b4a457dd/driver-db/ssd1306.toml#L13

this is however not true as it also has normal headers to wire it up (like most (all?) adafruit breakout boards).

what should be listed under `connection`? only "special" connectors like `StemmaQT` or also normal headers?
since this is only about the physical connection it gives no information about the protocol (I2C, SPI, etc.) - while the actual communication is most likely abstracted away by the driver it's still relevant since the MCU must support it; maybe that should be a separate field?

**Comment by tdittr** [comment]
Good point! I mainly wanted to list these connections so it is easy to search for scenarios like: "I want to play with embedded but I do not want to solder"

Feel free to add more connection options (I think headers can definitely one) :) same for the other enums, they only cover what I found while adding drivers to the list. Maybe I should make this clearer in the README.

When you add a new enum value just run the backend once and it will update the schema, also not ideal yet, but I did not have time to really round of that corner, see #39 

The linked SSD1306 board does have Stemma Qt according to their product name. A lot of the boards by Adafruit support that connector by now.

**Comment by rursprung** [comment]
> The linked SSD1306 board does have Stemma Qt according to their product name. A lot of the boards by Adafruit support that connector by now.

sorry, my sentence was maybe a bit misleading: yes, the linked board does have Stemma QT. but it also has headers. the "not true" was about it _only_ having Stemma QT as implied by the file.

for headers the question would be how you'd separate between proper headers (like with this board) or just cut-outs on the side (as is e.g. the case [on this board](https://www.digikey.ch/en/products/detail/ceva-technologies-inc/FSM300/7917164)) where you can't just put in headers but you can solder it to other pins? i'm not sure if there's already some established nomenclature for this (presumably there is and i'm just not aware of it)

**Comment by tdittr** [comment]
Mmh maybe we could be explicit that they are 2.54mm (0.1 inch) pitched holes? Since you still have to solder cables or headers to it? So something like `ThroughHoles100Thou`? Your other board could be `CastellatedHoles`

**Comment by tdittr** [comment]
Feel free to add now connector types to [`Connection`](https://github.com/tweedegolf/drive-rs/blob/d3703e1dec4856d7c5968bf08e24fd76091dfbad/backend/src/driver_db/boards.rs#L24) when time allows.
